app/models/pair_request.rb

```diff
@@ -83,4 +83,10 @@ def partner_for(user)
       author
     end
   end
+
+
+  scope :past, -> { where(status: 4) }
+  scope :upcoming, -> { where(status: 2) }
```

Comment:
I'm a bit curious why these are focused on the status at all? To my mind, it seems like this would just be a future/past query.